support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -51,24 +55,29 @@ class DigitalSubscriptionController(
 
   def digital(countryCode: String, orderIsAGift: Boolean): Action[AnyContent] = CachedAction() { implicit request =>
     implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    val title = if (orderIsAGift) {
-      "Support the Guardian | The Guardian Digital Gift Subscription"
-     } else {
-       "Support the Guardian | The Guardian Digital Subscription"
-     }
+    val title = s"Support the Guardian | The Guardian Digital ${if (orderIsAGift) "Gift " else ""}Subscription"
     val mainElement = EmptyDiv("digital-subscription-landing-page-" + countryCode)
     val js = Left(RefPath("digitalSubscriptionLandingPage.js"))
     val css = Left(RefPath("digitalSubscriptionLandingPage.css"))
     val description = stringsConfig.digitalPackLandingDescription
     val canonicalLink = Some(buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift))
-    val promoCodes: List[PromoCode] = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil) ++ DefaultPromotions.DigitalSubscription.all
+    val queryPromos = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
+    val promoCodes: List[PromoCode] = queryPromos ++ DefaultPromotions.DigitalSubscription.all
     val hrefLangLinks = Map(
       "en-us" -> buildCanonicalDigitalSubscriptionLink("us", orderIsAGift),
       "en-gb" -> buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift),
       "en-au" -> buildCanonicalDigitalSubscriptionLink("au", orderIsAGift),
       "en" -> buildCanonicalDigitalSubscriptionLink("int", orderIsAGift)
     )
-
+    val country = (for {
+      countryGroup <- CountryGroup.byId(countryCode)
+      country <- countryGroup.countries.headOption
+    } yield country).getOrElse(UK)
+    val maybePromotionCopy = queryPromos.headOption.flatMap(promoCode =>
+      ProductPromotionCopy(promotionServiceProvider
+        .forUser(false), stage)
```

Comment:
how do we use promo codes with UAT? if someone is a test user would they want to see UAT promo copy?

Reply:
I'm not sure about that- this is just how it's currently set up for the Guardian Weekly page.

Reply:
fair enough, I don't think it's essential that it works.